app/src/rooms/api/serializers.py

```diff
@@ -72,10 +72,19 @@ def create(self, validated_data):
                                    available_beds_double=available_beds_data.get("double"))
 
     def update(self, instance, validated_data):
-        beds_data = validated_data.pop("beds")
-        available_beds_data = validated_data.pop("available_beds")
+        beds_data = validated_data.pop("beds", {"single": instance.beds_single,
+                                                "double": instance.beds_double})
+        available_beds_data = validated_data.pop("available_beds",
+                                                 {"single": instance.available_beds_single,
+                                                  "double": instance.available_beds_double})
 
-        self._validate_beds(beds_data, available_beds_data)
+        beds_data_check = beds_data if beds_data \
+            else {"single": instance.beds_single, "double": instance.beds_double}
+        available_beds_data_check = available_beds_data if available_beds_data \
+            else {"single": instance.available_beds_single,
+                  "double": instance.available_beds_double}
```

Comment:
- What's the reason for this code? Why do you need to do 

`beds_data_check = beds_data if beds_data else {"single": instance.beds_single, "double": instance.beds_double}`

shouldn't that be the same as `beds_data_check = beds_data`? If so, why do you need this extra variable rather than using `beds_data` in the `_validate_beds` function?

- The other thing that I really don't like the fact that `beds` in the room serializer is a dict of two fields, while `Room` itself has just two separate fields.
- What's the difference between `beds` and `available_beds`?

Reply:
1) I have checked that beds_data can have three possible values: None, empty dict or dict with fields. This happens mostly when dealing with HTTP PUT method.
2) The value "beds" are real beds in the room, "available_beds" are beds we'd like to offer to guests.

Reply:
Then maybe it should look like this:
```
beds_data = validated_data.pop("beds") or {"single": instance.beds_single, "double": instance.beds_double})
```
But I think that more important would be to get rid of the convention of having a dict fields here